shell/src/main/java/alluxio/cli/docgen/MetricsDocGenerator.java

```diff
@@ -42,18 +44,48 @@
 @PublicApi
 public final class MetricsDocGenerator {
   private static final Logger LOG = LoggerFactory.getLogger(MetricsDocGenerator.class);
-  private static final String[] CATEGORIES =
-      new String[]{"cluster", "master", "worker", "client", "fuse"};
+  private static final String[] CATEGORIES = new String[]{
+      "client",
+      "cluster",
+      "fuse",
+      "master",
+      "worker",
+      };
   private static final String CSV_FILE_DIR = "docs/_data/table/";
   private static final String YML_FILE_DIR = "docs/_data/table/en/";
   private static final String CSV_SUFFIX = "csv";
   private static final String YML_SUFFIX = "yml";
   private static final String CSV_FILE_HEADER = "metricName,metricType";
+  private static final String TEMP_PREFIX = "temp-";
+
+  /**
+   * Compare the temp files and the committed files if validate flag is set.
+   *
+   * @param folder the location of the files to be compared
+   * @param suffix the file types (CSV/YML)
+   * @param category the category in the file names
+   * @param hasDiff a boolean indicating has there been changes
+   *
+   * @return a boolean indicating if there is anything difference
+   * between the temp files and committed files
+   */
+  public static boolean compareFiles(String folder, String suffix, String category,
```

Comment:
i would've expected this compareFiles to be very similar to the compareFiles in MetricsDocGenerator.java. i'm a bit confused why they are so different; can they be implemented similarly? being able to consolidate similar logic is important; for example, the problem below about deleting the temp file would not have been made if there was only one version of compareFiles